### PR TITLE
Feature add haser

### PR DESCRIPTION
**Pull request by mlohvynenko**

### CODE REVIEW

File: include/aos/common/crypto/crypto.hpp [hunk 1]
```diff
@@ -210,6 +210,52 @@ class HashType {
...
+ */
+class HasherItf {
+public:
+    /**
+     * Creates hash instance.
+     *
+     * @param algorithm hash algorithm.
+     * @return RetWithError<UniquePtr<HashItf>>.
+     */
+    virtual RetWithError<UniquePtr<HashItf>> CreateHasher(Hash algorithm) = 0;
```
Comment (al1img) [inline include/aos/common/crypto/crypto.hpp:251]: Should be named ad `CreateHash`  or `NewHash`.

File: src/common/crypto/mbedtls/cryptoprovider.cpp [hunk 2]
```diff
@@ -32,6 +32,75 @@ int mbedtls_x509_get_name(unsigned char** p, const unsigned char* end, mbedtls_x
...
+    /**
+     * Initializes hasher.
+     *
+     * @return Error.
+     */
+    aos::Error Init()
+    {
+        auto status = psa_hash_setup(&mOperation, mAlgorithm);
+        if (status != PSA_SUCCESS) {
+            LOG_ERR() << "Failed to begin hash operation: status=" << status;
```
Comment (al1img) [inline src/common/crypto/mbedtls/cryptoprovider.cpp:48]: Remove log

File: src/common/crypto/mbedtls/cryptoprovider.cpp [hunk 3]
```diff
@@ -32,6 +32,75 @@ int mbedtls_x509_get_name(unsigned char** p, const unsigned char* end, mbedtls_x
...
+
+            return aos::ErrorEnum::eFailed;
```
Comment (al1img) [inline src/common/crypto/mbedtls/cryptoprovider.cpp:50]: You can return error with message

File: src/common/crypto/mbedtls/cryptoprovider.cpp [hunk 4]
```diff
@@ -32,6 +32,75 @@ int mbedtls_x509_get_name(unsigned char** p, const unsigned char* end, mbedtls_x
...
+     * Updates hash with input data.
+     *
+     * @param data input data.
+     * @return Error.
+     */
+    aos::Error Update(const aos::Array<uint8_t>& data) override
+    {
+        auto status = psa_hash_update(&mOperation, data.begin(), data.Size());
+        if (status != PSA_SUCCESS) {
+            LOG_ERR() << "Failed to update hash operation: status=" << status;
```
Comment (al1img) [inline src/common/crypto/mbedtls/cryptoprovider.cpp:65]: ditto

File: src/common/crypto/mbedtls/cryptoprovider.cpp [hunk 5]
```diff
@@ -32,6 +32,75 @@ int mbedtls_x509_get_name(unsigned char** p, const unsigned char* end, mbedtls_x
...
+     */
+    aos::Error Finalize(aos::String& hash) override
+    {
+        size_t hashSize = 0;
+
+        aos::StaticArray<uint8_t, aos::cSHA256Size> buffer;
+
+        auto status = psa_hash_finish(&mOperation, buffer.begin(), buffer.MaxSize(), &hashSize);
+        if (status != PSA_SUCCESS) {
+            LOG_ERR() << "Failed to finish hash operation: status=" << status;
```
Comment (al1img) [inline src/common/crypto/mbedtls/cryptoprovider.cpp:87]: Ditto

File: tests/common/src/crypto_test.cpp [hunk 6]
```diff
@@ -918,3 +928,76 @@ TEST(CryptoTest, CreateUUIDv5)
...
+    ASSERT_NE(hasherPtr.Get(), nullptr);
+
+    const char* chunks[] = {
+        "",
+        "abc",
+        "string to test has works",
+    };
+
+    for (const auto& chunk : chunks) {
+
```
Comment (al1img) [inline tests/common/src/crypto_test.cpp:981]: Remove empty line

**Review by MykolaSolyanko — APPROVED** [review]

**Review by mykola-kobets-epam — APPROVED** [review]

**Pull request merged**